src/function/probability/lgamma.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-loss-of-precision */
+
+import { lgammaNumber, lnSqrt2PI } from '../../plain/number/index.js'
+import { deepMap } from '../../utils/collection.js'
+import { factory } from '../../utils/factory.js'
+
+const name = 'lgamma'
+const dependencies = ['Complex', 'typed']
+
+export const createLgamma = /* #__PURE__ */ factory(name, dependencies, ({ Complex, typed }) => {
+  const TWOPI = 6.2831853071795864769252842 // 2*pi
+  const LOGPI = 1.1447298858494001741434262 // log(pi)
+  const SMALLX = 7
+  const SMALLY = 7
+
+  const REFLECTION = 0.1
+
+  const coeffs = [
+    -2.955065359477124183e-2, 6.4102564102564102564e-3, -1.9175269175269175269e-3, 8.4175084175084175084e-4,
+    -5.952380952380952381e-4, 7.9365079365079365079e-4, -2.7777777777777777778e-3, 8.3333333333333333333e-2
+  ]
+
+  /**
+   * Logarithm of the gamma function for real, positive numbers.
+   * Based off of Tom Minka's lightspeed toolbox.
+   *
+   * Syntax:
+   *
+   *    math.lgamma(n)
+   *
+   * Examples:
+   *
+   *    math.lgamma(5)       // returns 3.178053830347945
+   *    math.lgamma(0)       // returns Infinity
+   *    math.lgamma(-0.5)    // returns NaN
+   *    math.lgamma(math.i)  // returns -0.6509231993018536 - 1.8724366472624294i
+   *
+   * See also:
+   *
+   *    gamma
+   *
+   * @param {number | Complex | Array | Matrix} n   A real or complex number
+   * @return {number | Complex | Array | Matrix}    The log gamma of `n`
+   */
+
+  return typed(name, {
+    number: lgammaNumber,
+
+    Complex: function (n) {
+      if (n.isNaN()) {
+        return new Complex(NaN, NaN)
+      } else if (n.im === 0) {
+        return new Complex(this(n.re), 0)
+      } else if (n.re >= SMALLX || Math.abs(n.im) >= SMALLY) {
+        return lgammaStriling(n)
+      } else if (n.re <= REFLECTION) {
+        // Reflection formula
+        const tmp = copysign(TWOPI, n.im) * Math.floor(0.5 * n.re + 0.25)
+        // TODO: `complex.js sin` doesn't have extremely high precision, so this value `a` may lose a little precision,
+        // causing the computation results to be less accurate than the lgamma of real numbers
+        const a = n.mul(Math.PI).sin().log()
+        const b = this(new Complex(1 - n.re, -n.im))
+        return new Complex(LOGPI, tmp).sub(a).sub(b)
+      } else if (n.im >= 0) {
+        return lgammaRecurrence(n)
+      } else {
+        return lgammaRecurrence(n.conjugate()).conjugate()
+      }
+    },
+
+    BigNumber: function () {
+      throw new Error("mathjs doesn't yet provide an implementation of the algorithm lgamma for BigNumber")
+    },
+
+    'Array | Matrix': function (n) {
+      return deepMap(n, this)
+    }
+  })
```

Comment:
We're no longer providing mapping methods on functions that also have different interpretations on square matrices, so please remove the Array/Matrix method (it can be recovered by the user with map(M, lgamma).

Reply:
Do we have any documentation talking about this?

Reply:
It's in the v11 branch, since it's a breaking change, but there's no point in putting in a matrix method now when it will just need to be taken out in the (imminent) v11 merge.

Reply:
Oh I see, thanks